language/move-vm/runtime/src/interpreter.rs

```diff
@@ -109,32 +106,13 @@ impl Interpreter {
         }
     }
 
-    /// Internal execution entry point.
-    fn execute(
-        &mut self,
-        loader: &Loader,
-        data_store: &mut impl DataStore,
-        gas_meter: &mut impl GasMeter,
-        extensions: &mut NativeContextExtensions,
-        function: Arc<Function>,
-        ty_args: Vec<Type>,
-        args: Vec<Value>,
-    ) -> VMResult<Vec<Value>> {
-        // No unwinding of the call stack and value stack need to be done here -- the context will
-        // take care of that.
-        self.execute_main(
-            loader, data_store, gas_meter, extensions, function, ty_args, args,
-        )
-    }
-
     /// Main loop for the execution of a function.
     ///
     /// This function sets up a `Frame` and calls `execute_code_unit` to execute code of the
     /// function represented by the frame. Control comes back to this function on return or
     /// on call. When that happens the frame is changes to a new one (call) or to the one
     /// at the top of the stack (return). If the call stack is empty execution is completed.
-    // REVIEW: create account will be removed in favor of a native function (no opcode) and
-    // we can simplify this code quite a bit.
+    /// This should only be called directly from `entrypoint()`, and `self` should not be used after it returns
     fn execute_main(
         &mut self,
```

Comment:
Based on the comment above, should this be `fn execute_main(self,` ?

Reply:
Good point, I can refactor to that.